SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::UpdateSettings(const FString& DevAuthToken)
+{
+	USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	GDKEditorSettings->SetDevelopmentAuthenticationToken(DevAuthToken);
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoGenerateDevAuthTokenTasks()
+{
+	bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, bIsRunningInChina]
+	{
+		AsyncTask(ENamedThreads::GameThread, [this]()
+		{
+			ShowTaskStartedNotification(TEXT("Generating Development Authentication Token"));
+		});
+
+		FString DevAuthToken;
+		FString ErrorMessage;
+		if (SpatialCommandUtils::GenerateDevAuthToken(bIsRunningInChina, DevAuthToken, ErrorMessage))
+		{
+			AsyncTask(ENamedThreads::GameThread, [this, DevAuthToken]()
```

Comment:
This GameThread switch is because of `EndTask` or `UpdateSettings`?
If `EndTask` requires `GameThread` then the function should switch itself

Reply:
No, I think it's clearer to have the async thread switching in one place. In this case, calling `EndTask` from a different thread would be a programmer error.